internal/driver/driver.go

```diff
@@ -152,6 +153,13 @@ func (d *Driver) Initialize(lc logger.LoggingClient, asyncCh chan<- *sdkModel.As
 		return errors.NewCommonEdgeXWrapper(edgexErr)
 	}
 
+	if d.config.AppCustom.EnableStatusCheck {
```

Comment:
need to wrap this with lock:

```suggestion
	d.configMu.RLock()
	enableStatusCheck := d.config.AppCustom.EnableStatusCheck
	d.configMu.RUnlock()
	
	if enableStatusCheck {
```